### PR TITLE
Akka Typed: Protocols

### PR DESCRIPTION
**Pull request by patriknw**

### CODE REVIEW

File: draft/protocols.md [hunk 1]
```diff
@@ -0,0 +1,54 @@
+---
+layout: series_post
+title: "Akka Typed: Protocols"
+description: ""
+author: Roland Kuhn & Patrik Nordwall
```
Comment (patriknw) [inline draft/protocols.md]: @rkuhn Do you mind if I add you as co-author of next blog post, since most of this is based on the ChatRoom example?

**Comment by rkuhn** [comment]
When shall it go live? I can only read it tonight, I fear.

**Comment by patriknw** [comment]
thursday or friday

File: draft/protocols.md [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+`ActorRef` that will be used to post messages. In this case we want to create a very simple Actor that just repackages the `PostMessage` command into a `PostSessionMessage` command which also includes the screen name. Such a wrapper Actor can be created by using the `spawnAdapter` method on the `ActorContext`, so that we can then go on to reply to the client with the `SessionGranted` result.
+
+The behavior that we declare here can handle both subtypes of `Command`. `GetSession` has been explained already and the `PostSessionMessage` commands coming from the wrapper Actors will trigger the dissemination of the contained chat room message to all connected clients. But we do not want to give the ability to send `PostSessionMessage` commands to arbitrary clients, we reserve that right to the wrappers we create—otherwise clients could pose as completely different screen names (imagine the `GetSession` protocol to include authentication information to further secure this). Therefore `PostSessionMessage` has ``private`` visibility and can't be created outside the actor.
```
Comment (rkuhn) [inline draft/protocols.md]: Perhaps mention that another strategy for exposing only part of the messages is to have a super-interface where no matching behavior of that type is publicly exposed. I agree that for Java `private` is probably better, but for Scala a sealed trait unlocks other benefits like exhaustiveness checking for the subset.

File: draft/protocols.md [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+From this behavior we can create an Actor that will accept a chat room session, post a message, wait to see it published, and then terminate.
+
+Now to try things out we must start both a chat room and a gabbler and we do that from the guardian supervisor:
+
+[TODO ChatRoom.scala root]
+
+([Same example in Java](https://github.com/patriknw/akka-typed-blog/blob/master/src/main/java/blog/typed/javadsl/ChatRoomApp.java))
+
+As illustrated in this example the typed `ActorRef` is a great tool for describing message protocols. Actor references with different types can be exchanged in the messages to describe the next type of messages that can be sent in an interaction.
```
Comment (rkuhn) [inline draft/protocols.md]: Perhaps add an outlook: «We are currently researching possibilities for deriving the message type definitions for a protocol from other sources, like a formal specification of [session types](http://groups.inf.ed.ac.uk/abcd/). First steps towards reusable and composable process steps for the implementing actors can be found at [akka-typed-session](https://github.com/rkuhn/akka-typed-session).»

**Comment by patriknw** [comment]
good additions, thanks @rkuhn 
It's live at http://blog.akka.io/typed/2017/05/12/typed-protocols

**Pull request merged**